sources/packages/web/src/components/common/ConfirmEnrolment.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <check-permission-role :role="Role.StudentConfirmEnrolment">
+    <template #="{ notAllowed }">
+      <v-btn
+        v-if="isConfirmCOEEnabled"
+        variant="text"
+        color="primary"
+        @click="submitConfirmEnrolment"
+        :disabled="notAllowed"
+        ><span class="text-decoration-underline"
+          ><strong>Confirm Enrolment</strong></span
+        ></v-btn
+      >
+    </template>
+  </check-permission-role>
+
+  <confirm-modal
+    title="Confirm enrolment"
+    ref="confirmEnrolmentModal"
+    okLabel="Confirm enrolment now"
+    cancelLabel="Cancel"
+    ><template #content
+      ><strong
+        >Are you sure you want to confirm enrolment for this
+        application?</strong
+      >
+      <p class="mt-5">
+        Confirming enrolment verifies this applicant is attending your
+        institution and will allow funding to be disbursed.
+      </p>
+    </template></confirm-modal
+  >
+</template>
+
+<script lang="ts">
+import { PropType, computed, defineComponent, ref } from "vue";
+import { COEStatus, Role, ApplicationStatus } from "@/types";
+import { ModalDialog } from "@/composables";
+import ConfirmModal from "@/components/common/modals/ConfirmModal.vue";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  emits: ["confirmEnrolment"],
+  components: { ConfirmModal, CheckPermissionRole },
+  props: {
+    coeStatus: {
+      type: Object as PropType<COEStatus>,
```

Comment:
If the enums are declared as `type: String as PropType<COEStatus>` they will be typed and Vue will not generate warnings. If I am not wrong, every place that we declared as `Object` it is generating a warning.